src/pages/Market/components/CoinList/CoinList.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+
+import CoinListStore from "@store/CoinListStore/CoinListStore";
+import { useQueryParamsStoreInit } from "@store/RootStore/hooks/useQueryParamsStoreInit";
+import rootStore from "@store/RootStore/instance";
+import { useLocalStore } from "@utils/useLocalStore";
+import { observer } from "mobx-react-lite";
+import ReactPaginate from "react-paginate";
+import { useSearchParams } from "react-router-dom";
+
+import Coins from "./components/Coins/Coins";
+import styles from "./styles.module.scss";
+
+const CoinList: React.FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const coinListStore = useLocalStore(() => new CoinListStore());
+  useQueryParamsStoreInit();
+
+  React.useEffect(() => {
+    const initialPage = rootStore.query.getParam("page");
+    if (initialPage) {
+      coinListStore.setItemOffset(
+        coinListStore.itemsPerPage * (Number(initialPage) - 1)
+      );
+    } else {
+      coinListStore.setItemOffset(0);
+    }
+
+    coinListStore.changePage();
+  }, []);
+
+  return (
+    <>
+      <Coins currentCoins={coinListStore.currentItems} />
+      <ReactPaginate
+        breakLabel="..."
+        nextLabel=">"
+        onPageChange={(event: { selected: number }) => {
+          coinListStore.handlePageClick(event);
+          searchParams.set("page", `${event.selected + 1}`);
+          setSearchParams(searchParams);
+        }}
+        forcePage={
+          rootStore.query.getParam("page") != undefined
```

Comment:
```suggestion
          rootStore.query.getParam("page") !== undefined
```
просто потому что строгость - залог успеха вот так вот просто да, а вообще это если что лишний раз ударит нас по рукам, если что-то потенциально пойдёт не так